task/run-opm-command-oci-ta/0.1/run-opm-command-oci-ta.yaml

```diff
@@ -0,0 +1,266 @@
+      type: string
+    - name: OPM_ARGS
+      description: The array of arguments to pass to the 'opm' command
+        (e.g., [ 'alpha', 'render-template', 'basic', 'v4.18/catalog-template.json']).
+      type: array
+    - name: OPM_OUTPUT_PATH
+      description: Relative path for the opm command's output file (e.g. 'v4.18/catalog/example-operator/catalog.json').
+        Relative to the root directory of given source code (Git repository).
+      type: string
+    - name: IDMS_PATH
+      description: Optional, path for ImageDigestMirrorSet file (e.g. '.tekton/images-mirror-set.yaml').
+      type: string
+      default: ""
+  results:
+    - name: SOURCE_ARTIFACT
+      description: The Trusted Artifact URI pointing to the artifact with
+        the application source code with generated file-based catalog from catalog-template.yml.
+  volumes:
+    - name: workdir
+      emptyDir: {}
+  stepTemplate:
+    volumeMounts:
+      - mountPath: /var/workdir
+        name: workdir
+  steps:
+    - name: use-trusted-artifact
+      image: quay.io/konflux-ci/build-trusted-artifacts:latest@sha256:635a890e9f0211b4eb5b8c811d6f06b3ed9ff7da952ebad236fbfe52ba36dbf7
+      args:
+        - use
+        - $(params.SOURCE_ARTIFACT)=/var/workdir/source
+    - name: run-opm-command
+      image: registry.redhat.io/openshift4/ose-operator-registry-rhel9:v4.18
+      workingDir: /var/workdir/source
+      env:
+        - name: OPM_OUTPUT_PATH_PARAM
+          value: $(params.OPM_OUTPUT_PATH)
+      securityContext:
+        runAsUser: 0
+      command: ["/bin/bash", "-c"]
+      args:
+        - |
+          set -euo pipefail
+
+          echo "Running OPM command in working directory: $(pwd)"
+          echo "OPM Arguments received by script: '$@'"
+
+          # Ensure OPM_OUTPUT_PATH_PARAM is provided
+          if [[ -z "${OPM_OUTPUT_PATH_PARAM}" ]]; then
+            echo "Error: OPM_OUTPUT_PATH is a mandatory parameter and cannot be empty."
+            exit 1
+          fi
+
+          # Validate that the path is not absolute
+          if [[ "${OPM_OUTPUT_PATH_PARAM}" == /* ]]; then
+            echo "Error: OPM_OUTPUT_PATH must be a relative path, but got '${OPM_OUTPUT_PATH_PARAM}'."
+            exit 1
+          fi
+
+          # Get the directory part of the path
+          OUTPUT_DIR=$(dirname "${OPM_OUTPUT_PATH_PARAM}")
+
+          # Create the directory if it's not the current directory '.'
+          if [[ "${OUTPUT_DIR}" != "." ]]; then
+            echo "Ensuring directory '${OUTPUT_DIR}' exists."
+            mkdir -p "${OUTPUT_DIR}"
+          fi
+
+          echo "Writing OPM output to file: $(pwd)/${OPM_OUTPUT_PATH_PARAM}"
+
+          # Execute the opm command and redirect its output
+          opm "$@" > "${OPM_OUTPUT_PATH_PARAM}"
+
+          echo "OPM command finished"
+        - "opm"
+        - "$(params.OPM_ARGS[*])"
+      computeResources:
+        limits:
+          memory: 2Gi
```

Comment:
Are you positive 2Gi memory will be sufficient for all OPM operations? Consider changing it to 4Gi just to err on the side of caution. 2Gi won't be sufficient if a product team wants to run OPM on index images